--- MPlayer-1.1.1.org/configure	2015-01-31 12:09:27.802844575 +0000
+++ MPlayer-1.1.1/configure	2015-01-31 12:10:24.814845418 +0000
@@ -803,7 +803,7 @@
 _struct_addrinfo=auto
 _getaddrinfo=auto
 _struct_sockaddr_storage=auto
-_smb=auto
+_smb=no
 _vidix=auto
 _vidix_pcidb=yes
 _dhahelper=no
@@ -4079,25 +4079,10 @@
 fi
 
 
-echocheck "Samba support (libsmbclient)"
-if test "$_smb" = yes; then
-  extra_ldflags="$extra_ldflags -lsmbclient"
-fi
-if test "$_smb" = auto; then
-  _smb=no
-  for ld_tmp in "-lsmbclient" "-lsmbclient $ld_dl" "-lsmbclient $ld_dl -lnsl" "-lsmbclient $ld_dl -lssl -lnsl" ; do
-    statement_check libsmbclient.h 'smbc_opendir("smb://")' $ld_tmp &&
-      extra_ldflags="$extra_ldflags $ld_tmp" && _smb=yes && break
-  done
-fi
-
-if test "$_smb" = yes; then
-    def_smb="#define CONFIG_LIBSMBCLIENT 1"
-    inputmodules="smb $inputmodules"
-else
-    def_smb="#undef CONFIG_LIBSMBCLIENT"
-    noinputmodules="smb $noinputmodules"
-fi
+echocheck "Samba support (libsmbclient) - disabled"
+_smb=no
+def_smb="#undef CONFIG_LIBSMBCLIENT"
+noinputmodules="smb $noinputmodules"
 echores "$_smb"
 
 
