diff -Naur libcap-2.25_orig/libcap/cap_file.c libcap-2.25/libcap/cap_file.c
--- libcap-2.25_orig/libcap/cap_file.c	2016-01-31 00:40:37.000000000 +0000
+++ libcap-2.25/libcap/cap_file.c	2017-11-30 01:17:25.910121396 +0000
@@ -7,6 +7,7 @@
 #include <sys/types.h>
 #include <byteswap.h>
 #include <sys/stat.h>
+#include <stddef.h>
 #include <unistd.h>
 #include <linux/xattr.h>
 
diff -Naur libcap-2.25_orig/libcap/cap_text.c libcap-2.25/libcap/cap_text.c
--- libcap-2.25_orig/libcap/cap_text.c	2013-12-16 04:46:28.000000000 +0000
+++ libcap-2.25/libcap/cap_text.c	2017-11-30 01:18:09.142120455 +0000
@@ -13,6 +13,7 @@
 #include "libcap.h"
 
 #include <ctype.h>
+#include <stddef.h>
 #include <limits.h>
 
 /* Maximum output text length (16 per cap) */
@@ -87,7 +88,7 @@
 #ifdef GPERF_DOWNCASE
 	const struct __cap_token_s *token_info;
 
-	token_info = __cap_lookup_name(str.constp, len);
+	token_info = __cap_lookup_name(str.constp, (size_t)len);
 	if (token_info != NULL) {
 	    *strp = str.constp + len;
 	    return token_info->index;
diff -Naur libcap-2.25_orig/libcap/Makefile libcap-2.25/libcap/Makefile
--- libcap-2.25_orig/libcap/Makefile	2016-01-31 00:01:41.000000000 +0000
+++ libcap-2.25/libcap/Makefile	2017-11-30 01:21:32.614116027 +0000
@@ -41,7 +41,7 @@
 	./_makenames > cap_names.h
 
 $(GPERF_OUTPUT): cap_names.list.h
-	perl -e 'print "struct __cap_token_s { const char *name; int index; };\n%{\nconst struct __cap_token_s *__cap_lookup_name(const char *, unsigned int);\n%}\n%%\n"; while ($$l = <>) { $$l =~ s/[\{\"]//g; $$l =~ s/\}.*// ; print $$l; }' < $< | gperf --ignore-case --language=ANSI-C --readonly --null-strings --global-table --hash-function-name=__cap_hash_name --lookup-function-name="__cap_lookup_name" -c -t -m20 $(INDENT) > $@
+	perl -e 'print "struct __cap_token_s { const char *name; int index; };\n%{\n#include <stddef.h>\n\nconst struct __cap_token_s *__cap_lookup_name(const char *, size_t);\n%}\n%%\n"; while ($$l = <>) { $$l =~ s/[\{\"]//g; $$l =~ s/\}.*// ; print $$l; }' < $< | gperf --ignore-case --language=ANSI-C --readonly --null-strings --global-table --hash-function-name=__cap_hash_name --lookup-function-name="__cap_lookup_name" -c -t -m20 $(INDENT) > $@
 
 cap_names.list.h: Makefile $(KERNEL_HEADERS)/linux/capability.h
 	@echo "=> making $@ from $(KERNEL_HEADERS)/linux/capability.h"
